openmc/lib/mesh.py

```diff
@@ -275,9 +295,184 @@ def set_grid(self, x_grid, y_grid, z_grid):
                                               ny, z_grid, nz)
 
 
+class CylindricalMesh(Mesh):
+    """CylindricalMesh stored internally.
+
+    This class exposes a mesh that is stored internally in the OpenMC
+    library. To obtain a view of a mesh with a given ID, use the
+    :data:`openmc.lib.meshes` mapping.
+
+    Parameters
+    ----------
+    index : int
+         Index in the `meshes` array.
+
+    Attributes
+    ----------
+    id : int
+        ID of the mesh
+    dimension : iterable of int
+        The number of mesh cells in each direction.
+    lower_left : numpy.ndarray
+        The lower-left corner of the structured mesh.
```

Comment:
Given that the other structured mesh implementations return these properties in cm, we may want to specify the units for the angular dimension here (same in the `openmc.lib.SphericalMesh`).